dlt/destinations/filesystem/filesystem.py

```diff
@@ -82,16 +83,22 @@ def __init__(self, schema: Schema, config: FilesystemDestinationClientConfigurat
         # verify files layout. we need {table_name} and only allow {schema_name} before it, otherwise tables
         # cannot be replaced and we cannot initialize folders consistently
         self.table_prefix_layout = path_utils.get_table_prefix_layout(config.layout)
-
-    @property
-    def dataset_path(self) -> str:
-        ds_path = posixpath.join(self.fs_path, self.config.normalize_dataset_name(self.schema))
-        return ds_path
+        self.dataset_path = posixpath.join(self.fs_path, self.config.normalize_dataset_name(self.schema))
 
     def drop_storage(self) -> None:
         if self.is_storage_initialized():
             self.fs_client.rm(self.dataset_path, recursive=True)
 
+    @contextmanager
+    def with_staging_dataset(self) -> Iterator["FilesystemClient"]:
+        current_dataset_path = self.dataset_path
+        try:
+            self.dataset_path = posixpath.join(self.fs_path, self.config.normalize_dataset_name(self.schema)) + "_staging"
```

Comment:
`self.config.normalize_dataset_name(SqlClientBase.make_staging_dataset_name(self.schema))`

Reply:
are you sure you want to use a method from the sqlclientbase in the filesystem destination? this does not seem quite right..

Reply:
so we need to move it to client base? or to `WithStagingDataset`?